File: python/src/aoc/year2018/day1.py
from aoc.util import load_input


def part1(lines):
    """
    >>> part1([+1, +1, +1])
    3
    >>> part1([+1, +1, -2])
    0
    >>> part1([-1, -2, -3])
    -6
    """
    return sum(int(i) for i in lines)


def part2(lines):
    """
    >>> part2([+1, -1])
    0
    >>> part2([+3, +3, +4, -2, -4])
    10
    >>> part2([-6, +3, +8, +5, -6])
    5
    >>> part2([+7, +7, -2, -7, -4])
    14
    """
    running_sum = 0
    memory = {0}
    while True:
        for value in lines:
            running_sum += int(value)
            if running_sum in memory:
                return running_sum
            memory.add(running_sum)


if __name__ == "__main__":
    data = load_input(__file__, 2018, '1')
    print(part1(data))
    print(part2(data))

File: kotlin/src/main/kotlin/io/github/ocirne/aoc/year2021/Day1.kt
package io.github.ocirne.aoc.year2021

import io.github.ocirne.aoc.AocChallenge

class Day1(lines: List<String>) : AocChallenge(2021, 1) {

    private val depths = lines.map { i -> i.toInt() }

    override fun part1(): Int {
        return depths.zipWithNext { a, b -> a < b }.count { it }
    }

    override fun part2(): Int {
        return depths
            .drop(2).indices
            .map { i -> depths.subList(i, i + 3).sum() }
            .zipWithNext { a, b -> a < b }
            .count { it }
    }
}
